File: python/day2_1.py
import sys
from collections import Counter

twos = 0
threes = 0

for l in sys.stdin:
    cnt = Counter()
    for c in l:
        cnt[c] += 1

    if any(c == 2 for c in cnt.values()):
        twos += 1
    if any(c == 3 for c in cnt.values()):
        threes += 1

print(twos * threes)

File: rust/src/bin/day2_1.rs
use std::io::prelude::*;
use std::io;

fn main() {
    let stdin = io::stdin();
    let (twos, threes) = stdin.lock()
        .lines()
        .map(|id| {
            let mut arr = [0; 26];
            for c in id.unwrap().chars() {
                let idx = c as usize - 'a' as usize;
                arr[idx] += 1;
            }

            arr
        })
        .fold((0, 0), |(twos, threes), map| {
            let twos = twos + if map.iter().any(|i| *i == 2) { 1 } else { 0 };
            let threes = threes + if map.iter().any(|i| *i == 3) { 1 } else { 0 };
            (twos, threes)
        });

    println!("CHECKSUM: {}", twos * threes);
}
